tests/MongoDB.Driver.Tests/Linq/Translators/FindProjectionTranslatorTests.cs

```diff
@@ -55,8 +55,10 @@ public void Should_not_translate_when_identity_is_present()
         public void Should_include_id_if_specified()
         {
             var result = Project(p => new { p.Id, p.A }, "{ _id: 1, A: \"Jack\" }");
+            var projection = result.Projection;
 
-            result.Projection.Should().Be("{ _id: 1, A: 1 }");
+            projection.GetValue("_id").Should().Be(1);
```

Comment:
Projection order was different on Linux, but values were correct.

Reply:
Odd.  Any hypothesis as to why there's a difference?

Reply:
we already had a similar issue here https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Legacy.Tests/Builders/UpdateBuilderTests.cs#L842 I think we even have a ticket for this. But I don't remember whether we investigated this or no

Reply:
@DmitryLukyanov I think the test you linked to involves the *server* returning fields in a different order.

The part of this test that is failing is executing client side only.

So I don't think they are not the same issue.

I would not have expected this projection to be different on Linux.

Reply:
@rstam, I don't remember all details, but I think the server response should not be related to the target framework which is used in the client app, shouldn't it?

Reply:
in fact, I think the server doesn't know anything about client TF or at least doesn't consider any logic for this, but maybe I'm missing something

Reply:
> but I think the server response should not be related to the target framework which is used in the client app

I believe that to be a true statement.

But that's not what's happening in this test. The Projection being asserted against is a client side result of translating a C# Expression to an MQL projection.

Reply:
Ok, I don't insist since I didn't investigate it.

Reply:
let me know if I need to make further investigation here

Reply:
We still don't understand why this change is necessary.

Maybe there is a hashtable involved somewhere deep inside the LINQ2 implementation that is returning entries in a different order on different platforms (so this only passed by coincidence before now)?

Reply:
>@DmitryLukyanov I think the test you linked to involves the server returning fields in a different order.

the mentioned legacy test is related to client-side behavior, the only reason why it fails without running a server is that we initialize a test collection in the constructor for each of the tests in this file, but this initialization is unrelated to the mentioned tests.
I don't say that this is the same issue, but I think they could have the same underlying reason

Reply:
Initializing a test collection should have no effect on the `Projection` which is a solely client side operation.

If the fields in the projection are really coming back out of order I think it is a driver bug.

Reply:
>Initializing a test collection should have no effect on the Projection which is a solely client side operation.

exactly what I wrote above :)

Reply:
https://jira.mongodb.org/browse/CSHARP-3284